dev/fattest.simplicity/src/componenttest/topology/impl/LibertyServer.java

```diff
@@ -5958,6 +6088,50 @@ public Set<String> getInstalledAppNames(String... possiblyInstalledAppNames) thr
         return subset;
     }
 
+    private static final String installFeatureMsgPrefix = "CWWKF0012I: The server installed the following features:";
+
+    /**
+     * Returns a set of the features which were installed at runtime, based on the messages.log.
+     *
+     * e.g.
+     * CWWKF0012I: The server installed the following features: [bells-1.0, cdi-4.0, componenttest-2.0, concurrent-3.0, jndi-1.0, mpConfig-3.1, mpContextPropagation-1.3,
+     * mpFaultTolerance-4.0, servlet-6.0, timedexit-1.0].
+     *
+     * @return           a set of the features installed at runtime
+     * @throws Exception
+     */
+    public Set<String> getInstalledFeatures() throws Exception {
+        Set<String> installedFeatures = new HashSet<>();
+
+        for (String line : findStringsInLogs(installFeatureMsgPrefix)) {
+            installedFeatures.addAll(getInstalledFeaturesFromLogMessage(line));
+        }
+
+        return installedFeatures;
+    }
+
+    /**
+     * Returns a set of the features which were installed at runtime, based on the CWWKF0012I message.
+     *
+     * e.g.
+     * CWWKF0012I: The server installed the following features: [bells-1.0, cdi-4.0, componenttest-2.0, concurrent-3.0, jndi-1.0, mpConfig-3.1, mpContextPropagation-1.3,
+     * mpFaultTolerance-4.0, servlet-6.0, timedexit-1.0].
+     *
+     * @return           a set of the features installed at runtime
+     * @throws Exception
+     */
+    public static final Set<String> getInstalledFeaturesFromLogMessage(String message) {
+        Set<String> installedFeatures = new HashSet<>();
+        String trimmed = message.substring(message.indexOf(installFeatureMsgPrefix)); //strip off the date & time etc
+        trimmed = trimmed.substring(trimmed.indexOf("[") + 1); //strip off everything up to and including the first "["
+        trimmed = trimmed.substring(0, trimmed.indexOf("]"));// remove "]."
+        String[] features = trimmed.split(", ");
```

Comment:
This will break in an ugly way if the message is not in the expected form since `indexOf` will return `-1`